aspnetcore/security/authorization/secure-data/samples/final3/Pages/Contacts/Create.cshtml.cs

```diff
@@ -54,7 +54,7 @@ public async Task<IActionResult> OnPostAsync()
                                                         ContactOperations.Create);
             if (!isAuthorized.Succeeded)
             {
-                return new ChallengeResult();
+                return Forbid();
```

Comment:
This one's ambiguous, we don't see if the user is authenticated because that check happens inside authorize. This should be split so that `if (!User.Identity.IsAuthenticated)` it calls Challenge, otherwise it calls Forbid. I already see something similar happening in the other pages.

Reply:
I didn't think it was possible to get into this action at all if the user wasn't authenticated, due to the `AuthorizeFilter` that's applied in `Startup`. Have I misunderstood something?

Reply:
This one? 
https://github.com/aspnet/AspNetCore.Docs/blob/6b3a072f7899b892b2f91d8e279f3f79257aee07/aspnetcore/security/authorization/secure-data/samples/final3/Startup.cs#L51-L54
You're right. Disregard.